Makefile.tests

```diff
@@ -0,0 +1,43 @@
+##
+# Run all the tests.
+#
+#  make run-tests
+#
+# run a selected suite of tests
+#
+#  make run-tests SUITE=tests/test_cosmology
+#
+# To debug a test
+#
+#  make tests/test_cosmology
+#  gdb tests/test_cosmology
+#
+##
+TESTED = \
+	interp \
+	powerspectrum \
+	cosmology \
+	timefac \
+	timebinmgr
+
+TESTBIN = $(TESTED:%=tests/test_%)
+SUITE?= $(TESTBIN)
+
+.PHONY: run-tests build-tests
+
+
+depends/lib/libcmocka.a:
+	cd depends; make lib/libcmocka.a
+
+build-tests: $(TESTBIN)
+
+run-tests: build-tests
+	for tt in $(SUITE) ; do $$tt ; done
+
+clean-tests:
+	rm -f $(TESTBIN)
+
+$(TESTBIN):
+
+tests/test_%: tests/test_%.c $(DESTDIR)/%.o $(DESTDIR)/endrun.o depends/lib/libcmocka.a
```

Comment:
If you link to endrun.o you need to call MPI_Init in main and MPI_Finalize in teardown for all the test modules.

Reply:
Yes. we shall certainly do that -- they have to deal with MPI at some point.